tests/utils/test_settings_utils.py

```diff
@@ -0,0 +1,164 @@
+import json
+import os
+from pathlib import Path
+from unittest import mock
+from django.test import TestCase, override_settings
+from django.conf import settings
+from arches.settings_utils import generate_frontend_configuration
+
+
+# these tests can be run from the command line via
+# python manage.py test tests.utils.test_settings_utils --settings="tests.test_settings"
+
+
+class TestFrontendConfigurationGeneration(TestCase):
+    """
+    Test the generation of frontend configuration files.
+    """
+
+    @mock.patch("arches.settings_utils.open")
+    @mock.patch("arches.settings_utils.json.dump")
+    @mock.patch("arches.settings_utils.list_arches_app_names")
+    @mock.patch("arches.settings_utils.list_arches_app_paths")
+    @mock.patch("arches.settings_utils.os.path.realpath")
+    def test_generate_frontend_configuration_core_arches(
+        self, mock_realpath, mock_app_paths, mock_app_names, mock_json_dump, mock_open
+    ):
+        """
+        Test frontend configuration generation when running core arches directly.
+        """
+        # Mock settings
+        mock_settings = mock.MagicMock()
+        mock_settings.APP_ROOT = "/arches/app"
+        mock_settings.ROOT_DIR = "/arches"
+        mock_settings.PUBLIC_SERVER_ADDRESS = "http://localhost:8000"
+        mock_settings.STATIC_URL = "/static/"
```

Comment:
I'd like to stick to the existing pattern for overriding settings during tests via the `@override_settings` decorator and thereby remove as much mocking as possible.
```suggestion
```

Reply:
applied